cmd/sqlflowserver/main_test.go

```diff
@@ -1598,6 +1606,116 @@ FROM housing.xgb_predict LIMIT 5;`)
 	}
 }
 
+var uniqueIDMutex sync.Mutex
+var uniqueID = 0
+
+func getUniqueID() int {
+	uniqueIDMutex.Lock()
+	defer uniqueIDMutex.Unlock()
+	uniqueID++
+	return uniqueID
+}
+
+func caseXGBoostFeatureColumnImpl(t *testing.T, table string, label string, selectColumns string, columnClauses string, nclasses int, nworkers int, isPai bool) {
+	tableSplits := strings.SplitN(table, ".", 2)
+	dbPrefix := ""
+	if len(tableSplits) == 2 {
+		dbPrefix = tableSplits[0] + "."
+	}
+
+	a := assert.New(t)
+	if columnClauses != "" {
+		columnClauses = "COLUMN " + columnClauses
+	}
+
+	trainSQLTemplate := `
+	SELECT %s FROM %s TO TRAIN xgboost.gbtree
+	WITH
+		objective="multi:softprob",
+		train.num_boost_round = 1,
+		train.num_workers = %d,
+		eta = 0.4,
+		num_class = %d,
+		validation.select="select %s from %s"
+	%s
+	LABEL %s
+	INTO %s;`
+
+	executeSQLFunc := func(sql string) {
+		_, _, _, err := connectAndRunSQL(sql)
+		a.NoError(err, fmt.Sprintf("SQL execution failure\n%s", sql))
+	}
+
+	// a unique id to avoid name conflict when run parallel
+	uniqueID := getUniqueID()
+
+	var modelName string
+	if isPai {
+		modelName = fmt.Sprintf("xgb_fc_test_model_%d", uniqueID)
+	} else {
+		modelName = fmt.Sprintf("%sxgb_fc_test_model_%d", dbPrefix, uniqueID)
+	}
+
+	trainSQL := fmt.Sprintf(trainSQLTemplate, selectColumns, table, nworkers, nclasses, selectColumns, table, columnClauses, label, modelName)
+	executeSQLFunc(trainSQL)
+
+	predictTableName := fmt.Sprintf("%sxgb_fc_test_predict_table_%d", dbPrefix, uniqueID)
+	predictSQL := fmt.Sprintf(`SELECT %s FROM %s TO PREDICT %s.%s_new USING %s;`, selectColumns, table, predictTableName, label, modelName)
+	executeSQLFunc(predictSQL)
+
+	if !isPai { // PAI does not support evaluate now
+		evaluateTableName := fmt.Sprintf("%sxgb_fc_test_evaluate_table_%d", dbPrefix, uniqueID)
+		evaluateSQL := fmt.Sprintf(`
+SELECT %s FROM %s TO EVALUATE %s WITH validation.metrics="accuracy_score" LABEL %s INTO %s;`,
```

Comment:
Join L1669 to L1668, just like L1663 does.

Reply:
Done.